src/services/schema-form.js

```diff
@@ -299,18 +327,30 @@ angular.module('schemaForm').provider('schemaForm',
 
       var stdForm = service.defaults(schema, ignore, options);
 
+      //ok let's merge!
+      //We look at the supplied form and extend it with schema standards
+      var lookup = stdForm.lookup;
+
       //simple case, we have a "*", just put the stdForm there
       var idx = form.indexOf('*');
       if (idx !== -1) {
-        form  = form.slice(0, idx)
-                    .concat(stdForm.form)
-                    .concat(form.slice(idx + 1));
+        // If there are more items in the form definition let's overwrite matches
+        if (form.length >= 1) {
+          form = form.slice(0, idx).concat(form.slice(idx + 1));
+          angular.forEach(lookup, function(k, v) {
```

Comment:
I'm not following why you do this? wouldn't it be easier to just loop over and extend the objects in the `lookup` map? They are the defaults from the schema and therefore the standard form.